src/Microsoft.ML.Transforms/StatefulCustomMappingTransformer.cs

```diff
@@ -0,0 +1,398 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Data;
+using Microsoft.ML.Internal.Utilities;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.Transforms
+{
+    /// <summary>
+    /// <see cref="ITransformer"/> resulting from fitting an <see cref="StatefulCustomMappingEstimator{TSrc, TState, TDst}"/>.
+    /// </summary>
+    /// <typeparam name="TSrc">The class defining which columns to take from the incoming data.</typeparam>
+    /// <typeparam name="TDst">The class defining which new columns are added to the data.</typeparam>
+    /// <typeparam name="TState">The type that describes per-cursor state.</typeparam>
+    public sealed class StatefulCustomMappingTransformer<TSrc, TDst, TState> : ITransformer
+        where TSrc : class, new()
+        where TDst : class, new()
+        where TState : class, new()
+    {
+        private readonly IHost _host;
+        private readonly Action<TSrc, TDst, TState> _mapAction;
+        private readonly Action<TState> _stateInitAction;
+        private readonly string _contractName;
+
+        internal InternalSchemaDefinition AddedSchema { get; }
+        internal SchemaDefinition InputSchemaDefinition { get; }
+
+        /// <summary>
+        /// Whether a call to <see cref="ITransformer.GetRowToRowMapper(DataViewSchema)"/> should succeed, on an
+        /// appropriate schema.
+        /// </summary>
+        bool ITransformer.IsRowToRowMapper => true;
```

Comment:
So 2 questions:

1) I see that `StatefulCustomMappingTransformer` doesn't implements the `IRowToRowMapper` interface (because it inherits from `RowToRowTransformBase`), so it doesn't have a `GetRow` method nor a `Row` class. According to what we chatted the other day, this would also mean that this transformer can't be used with `PredictionEngine`, is this right? I just want to make sure I am getting this correctly. I guess it makes sense since I guess `PredictionEngine` should only make predictions based on the contents of each row, and shouldn't have the ability to have a state.

2) Is it correct that this `ITransformer` is a `RowToRowMapper` whereas it returns a _Transform_ that actually doesn't implement IRowToRowMapper?